src/hotspot/os/linux/os_linux.cpp

```diff
@@ -217,6 +220,13 @@ julong os::available_memory() {
   return Linux::available_memory();
 }
 
+int os::Linux::malloc_info(FILE* stream) {
+  if (::malloc_info) {
+    return ::malloc_info(0, stream);
+  }
+  return 0;
```

Comment:
return -1.

You aim to mimic the malloc_info() API at os:: level, so you should return -1 in case malloc_info failed, including failure do to resolving issues.

Reply:
> return -1.
> 
> You aim to mimic the malloc_info() API at os:: level, so you should return -1 in case malloc_info failed, including failure do to resolving issues.

Wouldn't that lead to a caller not knowing whether `malloc_info` itself failed or if our wrapper failed? `malloc_info` also sets `errno`. I think it's better if we don't imitate `malloc_info`, by returning `-2` if `malloc_info` is missing, to distinguish the different states.

Reply:
> > > return -1.
> > > You aim to mimic the malloc_info() API at os:: level, so you should return -1 in case malloc_info failed, including failure do to resolving issues.
> > 
> > 
> > Wouldn't that lead to a caller not knowing whether `malloc_info` itself failed or if our wrapper failed?
> 
> Why would that matter?
> 
This matters when a caller sees -1, reads errno, and then takes an action depending on errno. If it was our wrapper that failed, then this will be a "stale" errno and a faulty action will be taken. This matters in `MallocInfoDcmd::execute` because we send different error messages depending on whether the wrapper failed or `malloc_info(3)` failed.

Reply:
Okay, fair enough. I would have thought the user cannot really do anything about it anyway, but your way is certainly more precise.